Fix Pruning

varunagrawal opened the pull request:
Update the return value of pruned factors/conditionals to return ∞ as the error in negative log space.

This way, when we normalize (by subtracting the minimum value) and take the exponential negative `exp(-e)`, we get `p=0` for pruned assignments, valid probabilities everywhere else.

With this fix, we can now run hybrid legged state estimation for 16 timesteps in 3.8 seconds, down from 13 minutes.

varunagrawal committed:
update pruning comment

varunagrawal committed:
take descriptive comment all the way

varunagrawal committed:
fix return value of pruned factors

dellaert reviewed (approved):
Very cool. Might want to update the PR comment with what you told me in email. Also, a test that fails before but works after would be great, so future PR’s don’t mess this up.

varunagrawal committed:
additional fix

varunagrawal committed:
clean up testHybridEstimation

varunagrawal committed:
more test improvements

varunagrawal committed:
add test for checking pruning

varunagrawal commented:
Added a test which originally fails, and found another subtle bug which I added a fix for.
Furthermore, improved `testHybridEstimation.cpp` quite a bit.
